core-bundle/src/Security/Voter/DataContainer/ArticleContentVoter.php

```diff
@@ -62,10 +62,11 @@ protected function hasAccessToRecord(TokenInterface $token, string $table, int $
 
     private function getPageId(int $articleId): int|null
     {
-        if (!isset($this->pageIds[$articleId])) {
-            $this->pageIds[$articleId] = $this->connection->fetchOne('SELECT pid FROM tl_article WHERE id=?', [$articleId]);
+        if (!array_key_exists($articleId, $this->pageIds)) {
+            $pid = $this->connection->fetchOne('SELECT pid FROM tl_article WHERE id=?', [$articleId]);
+            $this->pageIds[$articleId] = $pid ? (int) $pid : null;
```

Comment:
If we're at it, should we also harden the check to allow for `0` as a value?

```suggestion
            $this->pageIds[$articleId] = $pid !== false ? (int) $pid : null;
```

Reply:
`0` wouldn't be a valid return value for that method, it wouldn't be a page ID

Reply:
If it is fetched from the db it IMHO is. You can modify an INT PK to be 0 on db side without problems and it has no special meaning in Contao, does it?